src/stock-adjustment-mobile/edit-product-page/edit-product-page.jsx

```diff
@@ -0,0 +1,315 @@
+            }
+        }
+    }
+    ), []);
+
+    const isQuantityNotFilled = (quantity) => {
+        return _.isUndefined(quantity) || _.isNull(quantity) || _.isNaN(quantity) || quantity === "";
+    }
+
+    const validate = values => {
+        const errors = { items: [] };
+
+        values.items.forEach(item => {
+            let orderable = item.product;
+            if (!item.product) {
+                errors.items['product'] = { product: 'Required' };
+                orderable = [];
+            }
+
+            if (isQuantityNotFilled(item.quantity)) {
+                errors.items['quantity'] = { quantity: 'Required' };
+            }
+
+            if (!item.reason) {
+                errors.items['reason'] = { reason: 'Required' };
+            } else {
+                const stockOnHandQuantity = getStockOnHand(orderable, item?.lot?.lotCode ?? null);
+                if (!errors.items.hasOwnProperty('quantity')) {
+                    if (item.reason.reasonType !== "CREDIT" && item.quantity > stockOnHandQuantity) {
+                        errors.items['quantity'] = { quantity: 'Quantity cannot be greater than stock on hand value.' };
+                    }
+                }
+            }
+        });
+
+        return errors;
+    };
+
+    const getStockOnHand = (orderable, lotCode) => {
+        let returnedStock = null;
+        orderable.forEach(product => {
+            const productLotCode = product?.lot?.lotCode ?? null;
+            if (lotCode === productLotCode) {
+                returnedStock = product.stockOnHand;
+            }
+        });
+        return returnedStock;
+    };
+
+    const showToast = (type) => {
+        const toastPropertiesList = toastProperties.find((toast) => toast.title.toLowerCase() === type);
+        dispatch(setToastList([...toastList, toastPropertiesList]));
+    };
+
+    const cancel = () => {
+        history.goBack();
+    };
+
+    const deleteProduct = () => {
+        const adjustmentLenght = adjustment.length;
+        dispatch(setAdjustment(update(adjustment, { $splice: [[indexOfProductToEdit, 1]] } )));
+        showToast('success');
+        if (adjustmentLenght > 1) {
+            history.goBack();
+        }
+        else{
+            history.push('/makeAdjustmentAddProducts/submitAdjustment/programChoice');
+        }
+    };
+
+    const updateAdjustmentList = (values) => {
+        values.reasonFreeText = null;
+        values.occurredDate = formatDateISO(new Date());
+        values.reason = values.items[0].reason;
+        values.lot = values.items[0]?.lot ?? null;
+        values.displayLotMessage = values?.lot?.lotCode ?? "No lot defined";
+        values.quantity = values.items[0].quantity;
+
+        const productInformation = values.items[0].product;
+        const lotCode = values?.lot?.lotCode ?? null;
+        productInformation.forEach(prod => {
+            const productLotCode = prod?.lot?.lotCode ?? null;
+            if (lotCode === productLotCode) {
```

Comment:
Same as above, you may simplify to
```
if (lotCode === prod?.lot?.lotCode) {
```
Also, is there a case when `lotCode` and `productLotCode` are both `null`? Then the `if` statement would evaluate to `true`.